pkg/keys/ed25519_test.go

```diff
@@ -12,17 +18,66 @@ type Ed25519Suite struct{}
 var _ = Suite(&Ed25519Suite{})
 
 func (Ed25519Suite) TestUnmarshalEd25519(c *C) {
+	pub, _, _ := ed25519.GenerateKey(strings.NewReader("00001-deterministic-buffer-for-key-generation"))
+
+	publicKey, _ := json.Marshal(map[string]string{
+		"public": hex.EncodeToString(pub),
+	})
+	badKey := &data.PublicKey{
+		Type:       data.KeyTypeEd25519,
+		Scheme:     data.KeySchemeEd25519,
+		Algorithms: data.HashAlgorithms,
+		Value:      publicKey,
+	}
+	verifier := NewEd25519Verifier()
+	c.Assert(verifier.UnmarshalPublicKey(badKey), IsNil)
+}
+
+func (Ed25519Suite) TestUnmarshalEd25519_Invalid(c *C) {
 	badKeyValue, _ := json.Marshal(true)
 	badKey := &data.PublicKey{
-		Type:       data.KeyTypeRSASSA_PSS_SHA256,
-		Scheme:     data.KeySchemeRSASSA_PSS_SHA256,
+		Type:       data.KeyTypeEd25519,
+		Scheme:     data.KeySchemeEd25519,
 		Algorithms: data.HashAlgorithms,
 		Value:      badKeyValue,
 	}
-	verifier := NewP256Verifier()
+	verifier := NewEd25519Verifier()
 	c.Assert(verifier.UnmarshalPublicKey(badKey), ErrorMatches, "json: cannot unmarshal.*")
 }
 
+func (Ed25519Suite) TestUnmarshalEd25519_FastFuzz(c *C) {
+	verifier := NewEd25519Verifier()
+	for i := 0; i < 50; i++ {
+		// Ensure no basic panic
+
+		f := fuzz.New()
+		var publicData data.PublicKey
+		f.Fuzz(&publicData)
+
+		verifier.UnmarshalPublicKey(&publicData)
+	}
+}
+
+func (Ed25519Suite) TestUnmarshalEd25519_TooLongContent(c *C) {
+	randomSeed := make(json.RawMessage, 1024*1024)
+	io.ReadFull(rand.Reader, randomSeed)
```

Comment:
Why is this a `json.RawMessage` and not a `[]byte`? Additionally, could we check the `io.ReadFull` error?

The same things apply to `TestUnmarshalECDSA_TooLongContent`.

Reply:
json.RawMessage is a type alias to `[]byte`.

Reply:
Right, but `randomSeed` is being converted to a `data.HexBytes`, which is also an alias of `[]byte`. I don't see that `randomSeed` is used anywhere as a `json.RawMessage`.

Reply:
Fixed